azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -516,7 +523,20 @@ def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str]):
         self._auth_provider = _AadHelper(kcsb) if kcsb.aad_federated_security else None
         self._request_headers = {"Accept": "application/json", "Accept-Encoding": "gzip,deflate", "x-ms-client-version": "Kusto.Python.Client:" + VERSION}
 
+    @staticmethod
+    def _build_runtime_error(func_name: str) -> RuntimeError:
+        return RuntimeError(f"Kusto driver has detected a running event loop, consider using {func_name}_async instead of {func_name}")
+
     def execute(self, database: str, query: str, properties: ClientRequestProperties = None) -> KustoResponseDataSet:
+        """
+        Sync version of self.execute_async
+        """
+        try:
+            return async_to_sync(self.execute_async)(database, query, properties)
+        except RuntimeError as e:
+            raise self._build_runtime_error("execute") from e

```

Comment:
this means that running sync in a non-async env would break?

Reply:
Nope... it means that when trying to run sync flow from async env it would break...